/*
 * This file was automatically generated by EvoSuite
 * Mon Nov 30 13:12:56 GMT 2020
 */

package com.lts.io;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import com.lts.io.ArchiveScanner;
import com.lts.io.ImprovedFile;
import java.io.File;
import java.util.Vector;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.mock.java.io.MockFile;
import org.evosuite.runtime.testdata.EvoSuiteFile;
import org.evosuite.runtime.testdata.FileSystemHandling;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class ArchiveScanner_ESTest extends ArchiveScanner_ESTest_scaffolding {

  /**
  //Test case number: 0
  /*Coverage entropy=1.0905994737794786
  */
  @Test(timeout = 4000)
  public void test00()  throws Throwable  {
      EvoSuiteFile evoSuiteFile0 = new EvoSuiteFile("com.lts.io.ArchiveScanner/com.lts.io.ArchiveScanner/com.lts.io.ArchiveScanner0ory ");
      FileSystemHandling.createFolder(evoSuiteFile0);
      FileSystemHandling.createFolder(evoSuiteFile0);
      ImprovedFile improvedFile0 = new ImprovedFile("com.lts.io.ArchiveScanner", "com.lts.io.ArchiveScanner");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      String[] stringArray0 = new String[0];
      archiveScanner0.includes = stringArray0;
      Vector<Object> vector0 = new Vector<Object>();
      archiveScanner0.dirsNotIncluded = vector0;
      archiveScanner0.scandir(improvedFile0, "+p|w}RE({@j##RG%e1\":KP1V*J{H9(.jar", true);
      assertFalse(improvedFile0.isHidden());
  }

  /**
  //Test case number: 1
  /*Coverage entropy=0.6931471805599453
  */
  @Test(timeout = 4000)
  public void test01()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      String[] stringArray0 = new String[6];
      Vector<Object> vector0 = new Vector<Object>();
      archiveScanner0.filesNotIncluded = vector0;
      stringArray0[0] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      stringArray0[1] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      stringArray0[2] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      stringArray0[3] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      stringArray0[4] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      stringArray0[5] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      archiveScanner0.includes = stringArray0;
      archiveScanner0.processArchive(improvedFile0, "X1mUwS0X1mUwS.jar");
      assertFalse(improvedFile0.exists());
  }

  /**
  //Test case number: 2
  /*Coverage entropy=0.5623351446188083
  */
  @Test(timeout = 4000)
  public void test02()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      String[] stringArray0 = new String[6];
      Vector<Object> vector0 = new Vector<Object>();
      archiveScanner0.filesNotIncluded = vector0;
      stringArray0[0] = "co.lts.io.ArchiveScanne02#U.ja";
      ArchiveScanner archiveScanner1 = new ArchiveScanner(improvedFile0);
      archiveScanner1.includes = stringArray0;
      archiveScanner1.addDefaultExcludes();
      archiveScanner1.dirsNotIncluded = archiveScanner0.filesNotIncluded;
      // Undeclared exception!
      try { 
        archiveScanner1.processFile("co.lts.io.ArchiveScanne02#U.ja");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.lts.io.ArchiveScanner", e);
      }
  }

  /**
  //Test case number: 3
  /*Coverage entropy=0.5623351446188083
  */
  @Test(timeout = 4000)
  public void test03()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      String[] stringArray0 = new String[6];
      Vector<Object> vector0 = new Vector<Object>();
      archiveScanner0.filesNotIncluded = vector0;
      stringArray0[0] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      stringArray0[1] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      stringArray0[2] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      stringArray0[3] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      stringArray0[4] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      stringArray0[5] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      archiveScanner0.setExcludes(stringArray0);
      archiveScanner0.includes = archiveScanner0.excludes;
      ArchiveScanner archiveScanner1 = new ArchiveScanner(improvedFile0);
      archiveScanner1.setIncludes(archiveScanner0.includes);
      archiveScanner1.addDefaultExcludes();
      archiveScanner1.filesExcluded = archiveScanner0.filesNotIncluded;
      archiveScanner1.filesNotIncluded = archiveScanner1.filesExcluded;
      // Undeclared exception!
      try { 
        archiveScanner1.processFile("The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.lts.io.ArchiveScanner", e);
      }
  }

  /**
  //Test case number: 4
  /*Coverage entropy=0.5623351446188083
  */
  @Test(timeout = 4000)
  public void test04()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      String[] stringArray0 = new String[6];
      Vector<Object> vector0 = new Vector<Object>();
      archiveScanner0.filesNotIncluded = vector0;
      stringArray0[0] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      stringArray0[1] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      stringArray0[2] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      stringArray0[3] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      stringArray0[4] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      stringArray0[5] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      archiveScanner0.setExcludes(stringArray0);
      ArchiveScanner archiveScanner1 = new ArchiveScanner(improvedFile0);
      archiveScanner1.setExcludes(archiveScanner0.excludes);
      archiveScanner1.includes = stringArray0;
      archiveScanner1.dirsNotIncluded = archiveScanner0.filesNotIncluded;
      // Undeclared exception!
      try { 
        archiveScanner1.processFile("The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.lts.io.ArchiveScanner", e);
      }
  }

  /**
  //Test case number: 5
  /*Coverage entropy=0.5623351446188083
  */
  @Test(timeout = 4000)
  public void test05()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("J#UM]6Z6xvGXPdY^");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      String[] stringArray0 = new String[6];
      Vector<Object> vector0 = new Vector<Object>();
      archiveScanner0.filesNotIncluded = vector0;
      stringArray0[0] = "J#UM]6Z6xvGXPdY^";
      archiveScanner0.excludes = stringArray0;
      archiveScanner0.includes = stringArray0;
      // Undeclared exception!
      try { 
        archiveScanner0.processFile("J#UM]6Z6xvGXPdY^");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.lts.io.ArchiveScanner", e);
      }
  }

  /**
  //Test case number: 6
  /*Coverage entropy=0.5623351446188083
  */
  @Test(timeout = 4000)
  public void test06()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      String[] stringArray0 = new String[6];
      stringArray0[0] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      archiveScanner0.excludes = stringArray0;
      archiveScanner0.includes = stringArray0;
      Vector<Object> vector0 = new Vector<Object>();
      archiveScanner0.dirsExcluded = vector0;
      // Undeclared exception!
      try { 
        archiveScanner0.processFile("The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.lts.io.ArchiveScanner", e);
      }
  }

  /**
  //Test case number: 7
  /*Coverage entropy=0.5623351446188083
  */
  @Test(timeout = 4000)
  public void test07()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      String[] stringArray0 = new String[6];
      stringArray0[0] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      archiveScanner0.excludes = stringArray0;
      archiveScanner0.includes = stringArray0;
      Vector<Object> vector0 = new Vector<Object>();
      archiveScanner0.dirsExcluded = vector0;
      archiveScanner0.processDirectory(improvedFile0, "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.", true);
      assertFalse(improvedFile0.exists());
  }

  /**
  //Test case number: 8
  /*Coverage entropy=0.5623351446188083
  */
  @Test(timeout = 4000)
  public void test08()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("com.lts.io.ArchiveScanner.jar");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      String[] stringArray0 = new String[6];
      Vector<Object> vector0 = new Vector<Object>();
      archiveScanner0.filesNotIncluded = vector0;
      stringArray0[1] = "com.lts.io.ArchiveScanner.jar";
      stringArray0[2] = "com.lts.io.ArchiveScanner.jar";
      stringArray0[3] = "com.lts.io.ArchiveScanner.jar";
      stringArray0[4] = "com.lts.io.ArchiveScanner.jar";
      stringArray0[5] = "com.lts.io.ArchiveScanner.jar";
      stringArray0[0] = "co.lts.io.ArchiveScanne02#U.ja";
      archiveScanner0.excludes = stringArray0;
      archiveScanner0.includes = archiveScanner0.excludes;
      ArchiveScanner archiveScanner1 = new ArchiveScanner(improvedFile0);
      archiveScanner1.setIncludes(archiveScanner0.includes);
      archiveScanner1.addDefaultExcludes();
      archiveScanner1.filesExcluded = archiveScanner0.filesNotIncluded;
      archiveScanner1.filesNotIncluded = archiveScanner1.filesExcluded;
      // Undeclared exception!
      try { 
        archiveScanner1.processDirectory(improvedFile0, "co.lts.io.ArchiveScanne02#U.ja", true);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.lts.io.ArchiveScanner", e);
      }
  }

  /**
  //Test case number: 9
  /*Coverage entropy=0.5623351446188083
  */
  @Test(timeout = 4000)
  public void test09()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      String[] stringArray0 = new String[6];
      stringArray0[0] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      archiveScanner0.includes = stringArray0;
      archiveScanner0.addDefaultExcludes();
      Vector<Object> vector0 = new Vector<Object>();
      archiveScanner0.dirsExcluded = vector0;
      // Undeclared exception!
      try { 
        archiveScanner0.processDirectory(improvedFile0, "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.", true);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.lts.io.ArchiveScanner", e);
      }
  }

  /**
  //Test case number: 10
  /*Coverage entropy=1.3208883431493221
  */
  @Test(timeout = 4000)
  public void test10()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("com.lts.io.ArchiveScanner", "com.lts.io.ArchiveScanner");
      MockFile.createTempFile("Error trying to extract archive, ", "", (File) improvedFile0);
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      // Undeclared exception!
      try { 
        archiveScanner0.scandir(improvedFile0, "The system has encountered a severe, internal error", false);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.lts.io.DirectoryScanner", e);
      }
  }

  /**
  //Test case number: 11
  /*Coverage entropy=0.6931471805599453
  */
  @Test(timeout = 4000)
  public void test11()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      String[] stringArray0 = new String[6];
      Vector<Object> vector0 = new Vector<Object>();
      archiveScanner0.filesNotIncluded = vector0;
      stringArray0[0] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      stringArray0[1] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      stringArray0[2] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      stringArray0[3] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      stringArray0[4] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      stringArray0[5] = "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.";
      archiveScanner0.includes = stringArray0;
      archiveScanner0.processFile("|%Yh");
      assertEquals(1, ArchiveScanner.DIRECTORY);
  }

  /**
  //Test case number: 12
  /*Coverage entropy=0.6365141682948128
  */
  @Test(timeout = 4000)
  public void test12()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("J#UM]6Z6xvGXPdY^");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      String[] stringArray0 = new String[6];
      stringArray0[0] = "J#UM]6Z6xvGXPdY^";
      archiveScanner0.excludes = stringArray0;
      archiveScanner0.includes = stringArray0;
      Vector<String> vector0 = new Vector<String>();
      archiveScanner0.filesExcluded = vector0;
      archiveScanner0.processArchive(improvedFile0, "J#UM]6Z6xvGXPdY^");
      assertEquals(0L, improvedFile0.length());
  }

  /**
  //Test case number: 13
  /*Coverage entropy=0.6931471805599453
  */
  @Test(timeout = 4000)
  public void test13()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      int int0 = archiveScanner0.toFileType(improvedFile0);
      assertEquals(1, int0);
  }

  /**
  //Test case number: 14
  /*Coverage entropy=0.6931471805599453
  */
  @Test(timeout = 4000)
  public void test14()  throws Throwable  {
      ArchiveScanner archiveScanner0 = new ArchiveScanner((ImprovedFile) null);
      // Undeclared exception!
      try { 
        archiveScanner0.toFileType((File) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.lts.io.ArchiveScanner", e);
      }
  }

  /**
  //Test case number: 15
  /*Coverage entropy=0.6365141682948128
  */
  @Test(timeout = 4000)
  public void test15()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      archiveScanner0.scandir(improvedFile0, "", false);
      assertEquals(0L, improvedFile0.getFreeSpace());
  }

  /**
  //Test case number: 16
  /*Coverage entropy=0.6931471805599453
  */
  @Test(timeout = 4000)
  public void test16()  throws Throwable  {
      ArchiveScanner archiveScanner0 = new ArchiveScanner((ImprovedFile) null);
      // Undeclared exception!
      try { 
        archiveScanner0.scandir((File) null, (String) null, false);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.lts.io.ArchiveScanner", e);
      }
  }

  /**
  //Test case number: 17
  /*Coverage entropy=0.6931471805599453
  */
  @Test(timeout = 4000)
  public void test17()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      try { 
        archiveScanner0.scanArchive(improvedFile0, (String) null);
        fail("Expecting exception: Exception");
      
      } catch(Exception e) {
         //
         // Error trying to list archive, The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.
         //
         verifyException("com.lts.io.archive.AbstractNestedArchive", e);
      }
  }

  /**
  //Test case number: 18
  /*Coverage entropy=0.6931471805599453
  */
  @Test(timeout = 4000)
  public void test18()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("fCgj^iL", "fCgj^iL");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      try { 
        archiveScanner0.scanArchive(improvedFile0, "");
        fail("Expecting exception: Exception");
      
      } catch(Exception e) {
         //
         // Error trying to list archive, /home/ubuntu/termite/projects/78_caloriecount/fCgj^iL/fCgj^iL
         //
         verifyException("com.lts.io.archive.AbstractNestedArchive", e);
      }
  }

  /**
  //Test case number: 19
  /*Coverage entropy=0.6931471805599453
  */
  @Test(timeout = 4000)
  public void test19()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("J#UM]6Z6xvGXPdY^");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      String[] stringArray0 = new String[6];
      // Undeclared exception!
      try { 
        archiveScanner0.processFile(stringArray0[4]);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.lts.io.DirectoryScanner", e);
      }
  }

  /**
  //Test case number: 20
  /*Coverage entropy=0.6931471805599453
  */
  @Test(timeout = 4000)
  public void test20()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      // Undeclared exception!
      try { 
        archiveScanner0.processFile("");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.lts.io.DirectoryScanner", e);
      }
  }

  /**
  //Test case number: 21
  /*Coverage entropy=0.6931471805599453
  */
  @Test(timeout = 4000)
  public void test21()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      // Undeclared exception!
      try { 
        archiveScanner0.processDirectory(improvedFile0, (String) null, false);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.lts.io.DirectoryScanner", e);
      }
  }

  /**
  //Test case number: 22
  /*Coverage entropy=0.6931471805599453
  */
  @Test(timeout = 4000)
  public void test22()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      // Undeclared exception!
      try { 
        archiveScanner0.processDirectory((File) null, "", false);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.lts.io.DirectoryScanner", e);
      }
  }

  /**
  //Test case number: 23
  /*Coverage entropy=0.6931471805599453
  */
  @Test(timeout = 4000)
  public void test23()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("J#UM]6Z6xvGXPdY^");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      // Undeclared exception!
      try { 
        archiveScanner0.processArchive(improvedFile0, (String) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.lts.io.DirectoryScanner", e);
      }
  }

  /**
  //Test case number: 24
  /*Coverage entropy=0.6931471805599453
  */
  @Test(timeout = 4000)
  public void test24()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("comlts.io.rchiveScanner", "comlts.io.rchiveScanner");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      // Undeclared exception!
      try { 
        archiveScanner0.processArchive(improvedFile0, "");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.lts.io.DirectoryScanner", e);
      }
  }

  /**
  //Test case number: 25
  /*Coverage entropy=0.6931471805599453
  */
  @Test(timeout = 4000)
  public void test25()  throws Throwable  {
      ArchiveScanner archiveScanner0 = new ArchiveScanner((ImprovedFile) null);
      // Undeclared exception!
      try { 
        archiveScanner0.processArchive((File) null, "~x3ws");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.lts.io.DirectoryScanner", e);
      }
  }

  /**
  //Test case number: 26
  /*Coverage entropy=0.6931471805599453
  */
  @Test(timeout = 4000)
  public void test26()  throws Throwable  {
      ArchiveScanner archiveScanner0 = new ArchiveScanner((ImprovedFile) null);
      // Undeclared exception!
      try { 
        archiveScanner0.isArchive((File) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.lts.io.ArchiveScanner", e);
      }
  }

  /**
  //Test case number: 27
  /*Coverage entropy=0.6931471805599453
  */
  @Test(timeout = 4000)
  public void test27()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("fCgj^iL", "fCgj^iL");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      try { 
        archiveScanner0.scanArchive(improvedFile0, "fCgj^iL");
        fail("Expecting exception: Exception");
      
      } catch(Exception e) {
         //
         // Error trying to list archive, /home/ubuntu/termite/projects/78_caloriecount/fCgj^iL/fCgj^iL
         //
         verifyException("com.lts.io.archive.AbstractNestedArchive", e);
      }
  }

  /**
  //Test case number: 28
  /*Coverage entropy=0.6931471805599453
  */
  @Test(timeout = 4000)
  public void test28()  throws Throwable  {
      ArchiveScanner archiveScanner0 = new ArchiveScanner((ImprovedFile) null);
      // Undeclared exception!
      try { 
        archiveScanner0.scanArchive((File) null, "The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.lts.io.archive.AbstractNestedArchive", e);
      }
  }

  /**
  //Test case number: 29
  /*Coverage entropy=0.5623351446188083
  */
  @Test(timeout = 4000)
  public void test29()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("The system has encountered an severe error that has prevented it starting up.  This is probably the result of a bug or other problem that is beyond your control.  The details button may provide information that will help in resolving the problem so you may wish to copy the contents of that window and include it in any bug reports, etc.");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      boolean boolean0 = archiveScanner0.isArchive(improvedFile0);
      assertFalse(boolean0);
  }

  /**
  //Test case number: 30
  /*Coverage entropy=0.6931471805599453
  */
  @Test(timeout = 4000)
  public void test30()  throws Throwable  {
      ImprovedFile improvedFile0 = new ImprovedFile("X1mUwS0X1mUwS.jar");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      try { 
        archiveScanner0.scandir(improvedFile0, "te]pdi_0.jar", true);
        fail("Expecting exception: Exception");
      
      } catch(Exception e) {
         //
         // IO error scanning directory /home/ubuntu/termite/projects/78_caloriecount/X1mUwS0X1mUwS.jar
         //
         verifyException("com.lts.io.ArchiveScanner", e);
      }
  }

  /**
  //Test case number: 31
  /*Coverage entropy=1.0751393240053733
  */
  @Test(timeout = 4000)
  public void test31()  throws Throwable  {
      EvoSuiteFile evoSuiteFile0 = new EvoSuiteFile("com.lts.io.ArchiveScanner/com.lts.io.ArchiveScanner/com.lts.io.ArchiveScanner0ory ");
      FileSystemHandling.createFolder(evoSuiteFile0);
      ImprovedFile improvedFile0 = new ImprovedFile("com.lts.io.ArchiveScanner", "com.lts.io.ArchiveScanner");
      ArchiveScanner archiveScanner0 = new ArchiveScanner(improvedFile0);
      String[] stringArray0 = new String[0];
      archiveScanner0.includes = stringArray0;
      Vector<Object> vector0 = new Vector<Object>();
      archiveScanner0.dirsNotIncluded = vector0;
      archiveScanner0.scandir(improvedFile0, "+p|w}RE({@j##RG%e1\":KP1V*J{H9(.jar", false);
      assertTrue(improvedFile0.canRead());
  }
}
